### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by jacobcrawford**
The initial setup deploys a haproxy pod on all master nodes and load-balances traffic from one port to multiple backends.

### CODE REVIEW

File: chart/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        {{- include "services-haproxy.selectorLabels" . | nindent 8 }}
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+        livenessProbe:
+          failureThreshold: 8
+          httpGet:
+            host: localhost
+            path: /healthz
```
Comment (lynderup) [inline chart/templates/deployment.yaml:31]: Is this really the default path for liveness probe? I presume it is someone with madd skillz that choose that 😄

File: chart/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            port: {{ .Values.listenPort }}
+            scheme: HTTPS
+        volumeMounts:
+        - mountPath: /usr/local/etc/haproxy/haproxy.cfg
+          name: haproxyconf
+          subPath: haproxy.cfg
```
Comment (lynderup) [inline chart/templates/deployment.yaml:37]: Only use subpath if it is necessary. If you use subpath, changes in the config is only reflected in the pod if it is restarted, and argocd does not do that if the configmap is changed.

File: chart/values.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        matchExpressions: [
+        {
+          key: node-role.kubernetes.io/control-plane,
+          operator: Exists
+        }
+        ]
+      ]
+    }
+  }
+}
```
Comment (lynderup) [inline chart/values.yaml:32]: Why so many parentheses? Would it not be better to write it as normal yaml?

File: chart/templates/_helpers.tpl [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "services-haproxy.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create }}
+{{- default (include "services-haproxy.fullname" .) .Values.serviceAccount.name }}
+{{- else }}
+{{- default "default" .Values.serviceAccount.name }}
+{{- end }}
+{{- end }}
```
Comment (lynderup) [inline chart/templates/_helpers.tpl:62]: This is not used. Consider removing it.

**Review by lynderup — DISMISSED** [review]
Only minor comments. Not necessarily something that have to be changed

File: chart/scripts/wait_for_pod.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
+name=$1
+pod=0
+i=0
+pod_count=0
+echo "Waitin for pod $name"
+# Wait for the pod to exist
+while ([ 0 -eq $pod_count ] && [ $i -le 120 ]); do
+    i=$(($i + 1))
+    echo "Waiting for pod to exist "
+    kubectl get pods -A
```
[the remark this answers is not in the document]
Comment (jacobcrawford) [reply chart/scripts/wait_for_pod.sh:10]: purely debugging!

File: chart/templates/deployment.yaml (hunk 1 above)
Comment (jacobcrawford) [reply chart/templates/deployment.yaml:31]: I wish that it was my idea, but it is in the original guide:
https://github.com/kubernetes/kubeadm/blob/master/docs/ha-considerations.md#options-for-software-load-balancing
:D

**Review by lynderup — APPROVED** [review]

**Pull request merged**